fpga_interchange/fasm_generators/xc7/xc7.py

```diff
@@ -767,6 +792,377 @@ def handle_slice_ff(self):
             if init_value == 0:
                 self.add_cell_feature((tile_name, slice_site, bel, "ZINI"))
 
+    def get_cell_param(self, cell_data, name, force_format=None):
+        """
+        Retrieves definition and decodes value of a cell parameter. The
+        function can optionally force a specific string format if needed.
+        """
+
+        param = self.device_resources.get_parameter_definition(
+            cell_data.cell_type, name)
+
+        # Force the format if requested by substituting the paraameter
+        # definition object.
+        if not param.is_integer_like() and force_format is not None:
+            if force_format != param.string_format:
+                param = ParameterDefinition(
+                    name=name,
+                    string_format=force_format,
+                    default_value=cell_data.attributes[name])
+
+        return param.decode_integer(cell_data.attributes[name])
```

Comment:
This can be added to the generic FASM generator class I believe

Reply:
Moved to `utils.py`